New reference architecture diagram: Secure access to SaaS applications with SASE (#18818)
cloudflare/cloudflare-docs: src/content/docs/reference-architecture/diagrams/sase/secure-access-to-saas-applications-with-sase.mdx

SaaS applications are unable to evaluate the security posture of connecting devices. A compromised laptop with valid credentials appears identical to a securely managed, corporate device. When data is downloaded from the SaaS application, it has no visibility into where it goes or if the device it's being downloaded to is secure. Typically authentication for SaaS applications is externalized by redirecting users to an identity service, therefore the SaaS application has no sense of how the user authenticated and as such all trust is placed in the identity provider.

These security challenges are compounded by poor network access controls \- most SaaS applications accept connections from any Internet source, but sometimes they can be limited to only accessing from a specific set of IP addresses that might be associated with one or more physical offices. But these rudimentary network controls are hard to expand for remote users working from home, or partners and contractors who need access.

> Cloudflare's SASE platform offers the ability to bring a more Zero Trust orientated approach to securing SaaS applications. Centralized policies, based on device posture, identity attributes and granular network location can be applied across one or many SaaS applications. Cloudflare becomes the new corporate network, and it's possible to gate access to Internet based SaaS applications to those users and devices that are connected to Cloudflare. Essentially it's a new corporate network in the cloud.

Reviewer: ```suggestion
Cloudflare's SASE platform offers the ability to bring a more Zero Trust orientated approach to securing SaaS applications. Centralized policies, based on device posture, identity attributes and granular network location can be applied across one or many SaaS applications. Cloudflare becomes the new corporate network, and it is possible to gate access to Internet based SaaS applications to those users and devices that are connected to Cloudflare. Essentially it is a new corporate network in the cloud.
```

## Securing access with Cloudflare

The diagram below shows how Cloudflare sits between your users, devices and networks that require access to any SaaS application. The two main services proving security capabilities are:

> - [Zero Trust Network Access](/cloudflare-one/policies/access/). Allows Cloudflare to become an identity proxy. So that you can easily enable authentication with a wide variety of identity providers to a single SaaS application. This service also incorporates the ability to evaluate access based on device posture and network location.

Reviewer: ```suggestion
- [Zero Trust Network Access](/cloudflare-one/policies/access/). Allows Cloudflare to become an identity proxy, so that you can easily enable authentication with a wide variety of identity providers to a single SaaS application. This service also incorporates the ability to evaluate access based on device posture and network location.
```

> - [Secure Web Gateway](/cloudflare-one/policies/gateway/). Once all traffic to access the SaaS application flows through our gateway, HTTPS connections are terminated at Cloudflare and you have the ability to inspect the data flowing to and from the SaaS application. Blocking sensitive data from being exported to insecure locations.

Reviewer: ```suggestion
- [Secure Web Gateway](/cloudflare-one/policies/gateway/). Once all traffic to access the SaaS application flows through our gateway, HTTPS connections are terminated at Cloudflare and you have the ability to inspect the data flowing to and from the SaaS application.  This allows you to block sensitive data from being exported to insecure locations.
```

When integrating with an XDR platform such as Crowdstrike, Sentinel One or Microsoft InTune, device posture is also available for any authenticated user because Cloudflare matches the identity with the user in the XDR system and device posture information is evaluated.
:::

## Example policy

The following is an example set of policies which demonstrate how you can use Cloudflare to secure access to Salesforce.

The first step is using an [egress IP policy under Cloudflare Gateway](/cloudflare-one/policies/gateway/egress-policies/). This allows you to purchase and assign specific IPs to users that have their traffic filtered via Gateway. Then in Salesforce, you enforce that access is only permitted for traffic with a source IP that matches the one in your egress policy. This combination ensures that the only way to get access to Salesforce is via Cloudflare.

| Egress Policy                       |                  |
| :---------------------------------- | :--------------- |
| **Identity**                        |                  |
| User Group Names                    | All Employees    |
| **Select Egress IP**                |                  |

> | Use dedicated Cloudflare Egress IPs | \[203.0.113.88\] |

Reviewer: ```suggestion
| Use dedicated Cloudflare Egress IPs | `203.0.113.88` |
```

Reviewer: I've changed this to the standard way of displaying IP addresses. Not sure if the `[ ]` is relevant, though.

> This is important not only for securing access to Salesforce, but also for adequately protecting its contents while in use. Now let's look at the access policy that is limiting access to members of the Sales or Executives groups. We are also using our Crowdstrike integration to ensure that users are on company managed devices.

Reviewer: ```suggestion
This is important not only for securing access to Salesforce, but also for adequately protecting its contents while in use. Now let us examine the access policy that is limiting access to members of the Sales or Executives groups. We are also using our Crowdstrike integration to ensure that users are on company managed devices.
```

Reviewer: Adjusted the verb to ensure inclusive language.

| Gateway                        | On                                    |
| Crowdstrike Service to Service | Overall Score above 80                |

This second policy applies to all employees but we are going to apply a few more steps before access is granted.

| Policy name                    | Employees on trusted devices       |
| :----------------------------- | :--------------------------------- |
| Action                         | Allow                              |
| **Include**                    |                                    |
| Member of group                | All Employees                      |
| **Require**                    |                                    |
| Authentication method          | MFA \- multi-factor authentication |
| Gateway                        | On                                 |
| Crowdstrike Service to Service | Overall Score above 80             |
| **Additional Settings**        |                                    |
| Purpose justification          | On                                 |
| Temporary authentication       | On                                 |

> | Email addresses of approvers   | salesforce-admin@company.com       |

Reviewer: ```suggestion
| Email addresses of approvers   | `salesforce-admin@company.com`       |
```

Reviewer: To prevent this email from being clickable
